exercise/admin.py

```diff
@@ -23,6 +26,20 @@ class Meta:
 class ExerciseAdmin(admin.ModelAdmin):
     form = ExerciseAdminForm
     list_display = ("id", "title", "description", "length_minutes")
+    readonly_fields = ("copied_from",)
+    change_form_template = "admin/exercise/change_form.html"
+
+    def response_change(self, request, obj):
+        if "_copy_exercise" in request.POST:
+            copied_from = Exercise.objects.get(pk=obj.id)
+            obj.title = obj.title + " Copy"
+            obj.copied_from = copied_from
+            obj.id = None
+            obj.save()
+
+            self.message_user(request, f"Exercise {obj.title} created successfully.")
+            return HttpResponseRedirect(".")
```

Comment:
Let's redirect the user to the copy of the exercise so they can start to tweak that straight away.
```suggestion
            return redirect(reverse('admin:exercise_exercise_change', args=[obj.id]))
```

Reply:
Yes. Actually I forgot to ask about where this redirect should point. Nice catch